compiler/main.bal

```diff
@@ -61,19 +73,17 @@ public function main(string[] filenames, *Options opts) returns error? {
             return error(d:messageFormat(`unsupported extension ${ext}`));
         }
     }
+    if errorFileCount != 0 {
+        return error(string `compilation failed due to errors in ${errorFileCount} file(s)`);
```

Comment:
Don't use `file(s)`: do it properly.

Error message is not quite accurate: should say `compilation of N files failed`.

Reply:
fixed